fix(curriculum): ensure innerText test assertions use .trim()

nishatayub opened the pull request:
Checklist:

- [x] I have read and followed the contribution guidelines.
- [x] I have read and followed the how to open a pull request guide.
- [x] My pull request targets the main branch of freeCodeCamp.
- [ ] I have tested these changes locally.

This PR addresses an issue where some test assertions using `innerText` were failing due to newline characters or extra whitespace. By adding `.trim()`, we ensure consistent test behavior across environments.

For example, this assertion was failing:

```js
assert.strictEqual(span.innerText, 'Hello'); 
```

But due to an extra newline or space, the test would fail. This PR updates such assertions to:

```js
assert.strictEqual(span.innerText.trim(), 'Hello');
```

Related to #60690

<!-- Feel free to add any additional description of changes below this line -->

ilenia-magoni commented:
please describe what the issue is doing, your title is not clear

nishatayub commented:
> please describe what the issue is doing, your title is not clear, and you have not linked an issue
 just made the required changes, would love to know more

majestic-owl448 reviewed (commented):
please remove changes outisde curriculum/challenges/english/25-front-end-development/

and do not link to close the issue, but link the issue as related. While this PR catches many situations in which `trim` was needed, there are still many more (do not add them to the PR, it would become too big to review)

ilenia-magoni commented:
what are these last changes about?

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-hotel-feedback-form/66a962954f4e0d76223b37ed.md:
the last step is step 36, this change needs to be reverted

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-hotel-feedback-form/66a962954f4e0d76223b37ed.md:
````diff
@@ -56,13 +56,13 @@ assert.strictEqual(document.querySelectorAll("fieldset:nth-of-type(3) label + in
 
 ```js
-assert.strictEqual(document.querySelector('fieldset:nth-of-type(3) input[value="location"] + label')?.getAttribute('for'), 'reputation');
+assert.strictEqual(document.querySelectorAll('fieldset:nth-of-type(3) label')[3]?.getAttribute('for'), 'reputation');
````
these changes are outisde teh scope of the PR, what are these about?

please do not keep adding more changes, this PR is about adding `trim` only

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-hotel-feedback-form/66a962954f4e0d76223b37ed.md:
````diff
@@ -56,13 +56,13 @@ assert.strictEqual(document.querySelectorAll("fieldset:nth-of-type(3) label + in
 ```js
-assert.isNotNull(document.querySelectorAll('fieldset:nth-of-type(3) input[value="location"] + label')?.textContent, 'Reputation');
+assert.strictEqual(document.querySelectorAll('fieldset:nth-of-type(3) label')[3]?.textContent.trim(), 'Reputation');
 ```
 
 Your `label` should have the `for` attribute set to `"reputation"`.
 
 ```js
-assert.strictEqual(document.querySelector('fieldset:nth-of-type(3) input[value="location"] + label')?.getAttribute('for'), 'reputation');
+assert.strictEqual(document.querySelectorAll('fieldset:nth-of-type(3) label')[3]?.getAttribute('for'), 'reputation');
 ```
````
out of scope, please revert, and if you want to propose changes more than adding `trim`, create a new PR

nishatayub commented on curriculum/challenges/english/25-front-end-development/workshop-hotel-feedback-form/66a962954f4e0d76223b37ed.md:
````diff
@@ -56,7 +56,7 @@ assert.strictEqual(document.querySelectorAll("fieldset:nth-of-type(3) label + in
 
 ```js
-assert.isNotNull(document.querySelectorAll('fieldset:nth-of-type(3) input[value="location"] + label')?.textContent, 'Reputation');
+assert.isNotNull(document.querySelectorAll('fieldset:nth-of-type(3) input[value="location"] + label')?.textContent.trim(), 'Reputation');
````
Removed trim() from here

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-calorie-counter/63c9f24afbc9cf324dcaa9a4.md:
```diff
@@ -291,7 +291,7 @@ function clearForm() {
 
   budgetNumberInput.value = '';
-
+  output.innerText = '';
```
why did you change the seed code? here and in many other places

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-calorie-counter/63c9f28bda3e3f336e21b6b4.md:
```diff
@@ -291,7 +291,7 @@ function clearForm() {
 
   budgetNumberInput.value = '';
-  output.innerText = '';
+  output.innerText.trim()x = '';
```
here

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-decimal-to-binary-converter/645b6dad50514e69df601df6.md:
```diff
@@ -234,7 +234,8 @@ const decimalToBinary = (input) => {
 
   --fcc-editable-region--
-  
+  const binaryString = remainders.reverse().join("");
+  result.innerText = binaryString.trim();
```
other change to seed code

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-decimal-to-binary-converter/645c803cad8434791a5f57b6.md:
```diff
@@ -197,7 +197,7 @@ const result = document.getElementById("result");
 const decimalToBinary = (input) => {
   let binary = "";
-
+result.innerText = binary.trim()
```
other change to seed code

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-decimal-to-binary-converter/645c85251dc1827fe4a6c4ce.md:
```diff
@@ -199,7 +199,7 @@ const decimalToBinary = (input) => {
   }
 
-  result.innerText = binary;
+  result.innerText.trim() = binary;
```
other change to seed code

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-decimal-to-binary-converter/645c86f92a102d82250ce3e8.md:
```diff
@@ -209,7 +209,7 @@ const decimalToBinary = (input) => {
   }
 
-  result.innerText = binary;
+  result.innerText.trim() = binary;
```
other change to seed code

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-decimal-to-binary-converter/645c9e6cf5c7251f7b3308f6.md:
```diff
@@ -219,7 +219,7 @@ const decimalToBinary = (input) => {
   }
 
-  result.innerText = binary;
+  result.innerText.trim() = binary;
 };
```
other change to seed code

majestic-owl448 reviewed (commented):
there are changes to the seed code in many files, those changes are not appropriate, please remove them

I commented on a few, but there is more than that

nishatayub committed:
fix(curriculum): add .trim() to innerText/textContent test assertions

- Add .trim() to test assertions using innerText/textContent to handle whitespace
- Fix DOM selector in hotel feedback form step 24 test (issue #61635)
- Only modify test assertions in --hints-- sections, not seed code
- Ensures consistent test behavior across environments

Related to #60690

nishatayub committed:
fix(curriculum): add .trim() and Only modify test assertions, not seed code

nishatayub commented:
Changed all the changes i made to seed code.I wasn't mindful of the fact that we were focusing only on tests and ended up adding .trim() to all of innerText references.

nishatayub committed:
revert: restore original hotel feedback form test assertion

ilenia-magoni commented:
why don't you go look for some more instances that need `trim()` added? there were quite a lot of files when you first created this PR

nishatayub committed:
added trim() to more instances to innerText test assertions

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-hotel-feedback-form/66a84111965a0c46df6bbd0a.md:
````diff
@@ -50,8 +50,8 @@ assert.strictEqual(label?.getAttribute('for'), 'full-name');
 ```js
-const label = document.querySelector('fieldset legend + label');
-assert.strictEqual(label?.innerText, 'Name (required):');
+const label = document.querySelector('label');
+assert.strictEqual(label?.innerText.trim(), 'Name (required):');
````
test changes are to be discussed first, please stick only to adding `trim`

nishatayub replied:
Okay got it

nishatayub committed:
emoved test logic modifications

nishatayub commented on curriculum/challenges/english/25-front-end-development/workshop-balance-sheet/61fd7a160ed17960e971f28b.md:
````diff
@@ -28,8 +28,10 @@ Your `th` element should have the text `Total Assets`.
+const tableHeader = document.querySelector('th');
+assert.strictEqual(tableHeader.innerText.trim(), 'Total Assets');
+```
 ```
````
done

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-bookmark-manager-app/66def5467aee701733aaf8cc.md:
lines 150 and 258 can have the same treatment

ilenia-magoni replied:
not done

nishatayub replied:
okay done. Let me know if there is more.

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-event-hub/66ebd4ae2812430bb883c787.md:
line 251 and 259 could also need `trim()`

nishatayub replied:
done

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lab-sorting-visualizer/6716249b5405164036fd0b0d.md:
lines 204 217 224 278 could also need `trim()`

ilenia-magoni replied:
not done

nishatayub replied:
just Done

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-balance-sheet/61fd7a160ed17960e971f28b.md:
lines 40 64 72 80 could also need `trim()`

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-blog-page/66a338be7f1dee383a0e0ecb.md:
lines 52 61 could also need `trim()`

nishatayub replied:
done

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-blog-page/66a33a00514b40393a983c24.md:
lines 52 61 could also need `trim()`

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-calorie-counter/63c9e8fe3a6f022a05a04675.md:
line 209 could also need `trim()`

nishatayub replied:
done

majestic-owl448 reviewed (commented):
there are a few more instances in the same files of `innerText` and `textContent` you can add `trim` to

nishatayub committed:
trim() added to areas requested

nishatayub committed:
added trim() to other files as per requested by the reviewer

nishatayub commented:
Hey, just wanted to let you know all requested changes have been made and the checks have passed. Let me know if anything else is needed.

ilenia-magoni commented:
You may have missed this 
<img width="414" height="62" alt="image" src="https://github.com/user-attachments/assets/d280d541-e4f3-4077-aef4-d472187aa8b8" />

You will need to be patient for a second reviewer

nishatayub commented:
Oh Sure

moT01 reviewed (approved):
LGTM @nishatayub 🎉 Congratulations on your first contribution to the freeCodeCamp repo 🎉
